plateau/core/common_metadata.py

```diff
@@ -746,14 +752,29 @@ def empty_dataframe_from_schema(schema, columns=None, date_as_object=False):
     columns: Union[None, List[str]]
         Optional list of columns that should be part of the resulting DataFrame. All columns in that list must also be
         part of the provided schema.
+    date_as_object: bool
+        Cast dates to objects.
+    coerce_temporal_nanoseconds: bool
+        Coerce date32, date64, duration and timestamp units to nanoseconds to retain behaviour of pandas 1.x.
+        Only applicable to pandas version >= 2.0 and PyArrow version >= 13.0.0.
 
     Returns
     -------
     DataFrame
         Empty DataFrame with requested columns and types.
     """
+    # HACK: Cast bytes to object in metadata until Pandas bug is fixed: https://github.com/pandas-dev/pandas/issues/50127
+    schema = schema_metadata_bytes_to_object(schema.internal())
 
-    df = schema.internal().empty_table().to_pandas(date_as_object=date_as_object)
+    if PYARROW_LT_13:
+        # Prior to pyarrow 13.0.0 coerce_temporal_nanoseconds didn't exist
+        # as it was introduced for backwards compatibility with pandas 1.x
+        df = schema.empty_table().to_pandas(date_as_object=date_as_object)
+    else:
+        df = schema.empty_table().to_pandas(
+            date_as_object=date_as_object,
+            coerce_temporal_nanoseconds=coerce_temporal_nanoseconds,
+        )
```

Comment:
```suggestion
    _coerce = {}
    if PYARROW_LT_13:
        # Prior to pyarrow 13.0.0 coerce_temporal_nanoseconds didn't exist
        # as it was introduced for backwards compatibility with pandas 1.x
        _coerce["coerce_temporal_nanoseconds"] = coerce_temporal_nanoseconds
        
    df = schema.empty_table().to_pandas(date_as_object=date_as_object, **_coerce)
```
Would this be slightly nicer? It reduces code duplication a tiny bit (same below).

Reply:
Yeah, it looks to be a neater way of doing things, but I think the if statement is the wrong way around - we want to provide the coerce argument if pyarrow>=13 since in pyarrow<13 coercion is the default/only behaviour. So I could either switch the if statement to be `if not PYARROW_LT_13` or else redefine the boolean as `PYARROW_GTE_13` - do you have a preference?

Reply:
If you're a fan of one-liners:
```
_coerce = {} if PYARROW_LT_13 else {"coerce_temporal_nanoseconds": coerce_temporal_nanoseconds}
```
I'm not necessarily convinced that this is any cleaner, so it doesn't matter to me either way

Reply:
Right now I've gone for the one-liner wherever it fits, and `if not PYARROW_LT_13` wherever the one liner would be forced onto multiple lines by pre-commit. Let me know if you'd like me to change it.